dapps/marketplace/src/pages/growth/Promotions.js

```diff
@@ -0,0 +1,289 @@
+import React from 'react'
+import { fbt } from 'fbt-runtime'
+import Link from 'components/Link'
+import { withRouter } from 'react-router-dom'
+import withWallet from 'hoc/withWallet'
+
+import ActionList from 'components/growth/ActionList'
+import MobileModalHeader from 'components/MobileModalHeader'
+import ShareableContent from 'components/growth/ShareableContent'
+
+import { Mutation } from 'react-apollo'
+import VerifyPromotionMutation from 'mutations/VerifyPromotion'
+import AutoMutate from 'components/AutoMutate'
+
+import { formatTokens, getContentToShare } from 'utils/growthTools'
+
+import get from 'lodash/get'
+
+const getToastMessage = (action, decimalDivision) => {
+  const tokensEarned = formatTokens(action.reward.amount, decimalDivision)
+  return fbt(
+    `You earned ` + fbt.param('amount', tokensEarned) + ' OGN',
+    'GrowthPromotions.tokensEarned'
+  )
+}
+
+const actionTypeToNetwork = actionType => {
+  switch (actionType) {
+    case 'TwitterShare':
+      return 'TWITTER'
+  }
+
+  return null
+}
+
+const getActionHash = action =>
+  web3.utils.sha3(`${action.type}/${action.content.post.text.default}`)
+
+class Promotions extends React.Component {
```

Comment:
Can we use a bunch of separate functional components here instead of a Component with lots of different renderX methods? Hoping we can move away from React.Component in new code

Reply:
If it's a lot of work to change then no need.. but bear in mind for the future

Reply:
Thanks. Will do that.

Reply:
@nick Does this look good? https://github.com/OriginProtocol/origin/commit/13bf837576f15dea4688f81fbfa4117538d1c892